inbox/instrumentation.py

```diff
@@ -112,13 +112,21 @@ def __init__(
         # thread.
         self.statsd_client = get_statsd_client()
         self.start_time = time.time()
+        self._stopping = False
+        self._thread_id = None
 
     def start(self):
         self.start_time = time.time()
         greenlet.settrace(self._trace)
         # Spawn a separate OS thread to periodically check if the active
         # greenlet on the main thread is blocking.
-        gevent._threading.start_new_thread(self._monitoring_thread, ())
+        self._thread_id = gevent._threading.start_new_thread(
+            self._monitoring_thread, ()
+        )
+
+    def stop(self):
```

Comment:
This is just to be nice and not leave this thread running for the rest of the test suite. I cannot `.join()` it because I don't have a real `threading.Thread` object, just a raw OS thread_id.